libcperciva/util/sock_util.c

```diff
@@ -335,3 +336,36 @@ sock_addr_ensure_port(const char * addr)
 	/* Failure! */
 	return (NULL);
 }
+
+/**
+ * sock_addr_validate(addr):
+ * Check that ${addr} is syntactically valid, but do not perform any address
+ * resolution.
+ */
+int
+sock_addr_validate(const char * addr)
+{
+
+	/* Sanity checks. */
+	if (addr == NULL) {
+		warn0("Empty sock address.");
+		goto err0;
+	}
+	if (strlen(addr) == 0) {
+		warn0("Empty sock address.");
+		goto err0;
+	}
+
+	/* Check for a missing hostname. */
```

Comment:
UNIX socket addresses don't have hostnames.  The code is actually fine since UNIX socket addresses will start with `/` so anything with `addr[0] == ':'` isn't a valid UNIX *or* network socket address, but the comment (and possibly the error message) needs to be clarified.